### ISSUE TITLE
setup environment for multi-level testing

### ISSUE THREAD

**Issue by cjdcordeiro** [mF2C/mF2C#64]
setup with a leader and children, and with DC replication

**Comment by rosogon** [comment]
Hi @cjdcordeiro , @alexbarcelo , @aqueralt 

I am trying to setup a leader agent and a child agent (discovery not needed, just dataclay). I am using https://github.com/mF2C/mF2C/blob/master/docker-compose/docker-compose.yml. I commented out compss, discovery, policies, cau-client, identification, user-management, resource-categorization, snaptemp, influxdb, snap, influxdb-analytics, analytics_engine, neo4j, landscaper and web. 

For the moment, I am just starting up the leader agent. When I docker-compose up, there is an error with dcproxy and it dies:

```dcproxy_1          | Exception in thread "main" java.lang.NoSuchMethodError: dataclay.commonruntime.ClientManagementLib.registerExternalDataClay(Ljava/lang/String;I)Ldataclay/util/ids/DataClayInstanceID;
dcproxy_1          |    at api.DataClayWrapper.init(DataClayWrapper.java:73)
dcproxy_1          |    at com.sixsq.dataclay.proxy$_main$fn__1019.invoke(proxy.clj:39)
dcproxy_1          |    at com.sixsq.dataclay.proxy$_main.invokeStatic(proxy.clj:37)
dcproxy_1          |    at com.sixsq.dataclay.proxy$_main.doInvoke(proxy.clj:28)
dcproxy_1          |    at clojure.lang.RestFn.invoke(RestFn.java:397)
dcproxy_1          |    at clojure.lang.AFn.applyToHelper(AFn.java:152)
dcproxy_1          |    at clojure.lang.RestFn.applyTo(RestFn.java:132)
dcproxy_1          |    at com.sixsq.dataclay.proxy.main(Unknown Source)
dockercompose_dcproxy_1 exited with code 1
```


Anyway,  I can GET /api/agreement, f.e. But I cannot post. The error is:

```{
  "status" : 405,
  "message" : "invalid method (post) for /api/agreement",
  "resource-id" : "/api/agreement"
}
```

I guess the second issue is related to the first, so let's concentrate on the dcproxy error first.

**Comment by rosogon** [comment]
Docker-compose file here:
https://github.com/mF2C/mF2C/blob/feature/sla-dataclay/docker-compose/docker-compose.yml

**Comment by cjdcordeiro** [comment]
by looking at the exception it looks like a compatibility issue, as it cannot find a specific method. @rosogon are you using the `:latest` images?

@alexbarcelo can you replicate? and do you know which method that's referring to?

**Comment by cjdcordeiro** [comment]
@rosogon try again. I've added latest to cimi.

also make sure you're re-pulling the latest tags

**Comment by rosogon** [comment]
I removed every image using `docker images -q | xargs docker rmi`, and then a `docker-compose pull`.

I realized that dcproxy is failing because it has CHILDREN_DC set  (https://github.com/mF2C/mF2C/blob/feature/sla-dataclay/docker-compose/docker-compose.yml#L24). I commented out the env var and now is failing at POST time, but not dying.

Stack trace:

```
SEVERE: error executing request: [:add {:description "Traffic Lights Emergency Service", :category {:inclinometer false, :disk "low", :ir_motion false, :pump_sensor false, :door_sensor false, :memory "low", :air_pressure false, :humidity false, :jammer false, :accelerometer false, :storage "low", :network "low", :battery_level false, :location false, :temperature false, :cpu "low"}, :exec "upcuc1/uc1-emergency3:latest", :updated "2018-07-27T13:48:06.988Z", :name "emergency-traffic-demo", :created "2018-07-27T13:48:06.988Z", :exec_type "docker", :id "service/ba671e02-11e0-4148-9dcf-e27e559826f5", :acl {:owner {:principal "ADMIN", :type "ROLE"}}, :exec_ports [], :resourceURI "http://schemas.dmtf.org/cimi/2/Service"} {}]
dcproxy            | java.lang.ArithmeticException: / by zero
dcproxy            | java.lang.ArithmeticException: / by zero
dcproxy            |    at dataclay.commonruntime.DataClayRuntime.getExecutionLocationIDFromHash(Unknown Source)
dcproxy            |    at dataclay.commonruntime.DataClayRuntime.makePersistent(Unknown Source)
dcproxy            |    at dataclay.DataClayObject.makePersistent(Unknown Source)
dcproxy            |    at api.DataClayWrapper.create(DataClayWrapper.java:223)
dcproxy            |    at com.sixsq.dataclay.scrud$add.invokeStatic(scrud.clj:39)
dcproxy            |    at com.sixsq.dataclay.scrud$add.invoke(scrud.clj:35)
dcproxy            |    at clojure.lang.AFn.applyToHelper(AFn.java:156)
dcproxy            |    at clojure.lang.AFn.applyTo(AFn.java:144)
dcproxy            |    at clojure.core$apply.invokeStatic(core.clj:657)
dcproxy            |    at clojure.core$apply.invoke(core.clj:652)
dcproxy            |    at com.sixsq.dataclay.handler$dispatch.invokeStatic(handler.clj:64)
dcproxy            |    at com.sixsq.dataclay.handler$dispatch.invoke(handler.clj:52)
dcproxy            |    at com.sixsq.dataclay.handler$handler.invokeStatic(handler.clj:75)
dcproxy            |    at com.sixsq.dataclay.handler$handler.invoke(handler.clj:72)
dcproxy            |    at org.httpkit.server.HttpHandler.run(RingHandler.java:91)
dcproxy            |    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
dcproxy            |    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
dcproxy            |    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
dcproxy            |    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
dcproxy            |    at java.lang.Thread.run(Thread.java:748)
```

I think you both can easily test this by git clone ... && git checkout feature/sla-dataclay && docker-compose up

**Comment by cjdcordeiro** [comment]
`dataclay.commonruntime.DataClayRuntime.getExecutionLocationIDFromHash(Unknown Source)` I'm afraid I can't help much with this

**Comment by alexbarcelo** [comment]
There was at least a problem in the usage of a deprecated environment variable name. That is solved (minor commit done sparsely to fix `DS` environment files) and now te DataService starts, but there is still something wrong. We are looking into it.

**Comment by alexbarcelo** [comment]
Sorry, this issue slipped under my radar.

I am looking into the commit history of mF2C/dataClay, and during July 30th and 31st all the upgrade and fixes were commited (+ some minor overlooks, solved in August 6th).

It should be fixed by then. I am not closing this just in case, I remember doing "multi-level" sanity checks mid-august... but after the holidays, maybe there is still missing something.

**Comment by rosogon** [comment]
@alexbarcelo : The previous error is fixed. I can post an agreement when in leader. I can also retrieve it. 

I will test now with the child

**Comment by cjdcordeiro** [comment]
done. waiting on @rosogon . if it works, then push to D Hub and close this

**Comment by aqueralt** [comment]
Roman just confirmed. We will push the new version today-tomorrow. However, he found a different issue about some updates that are happening from time to time that he seems not to be calling. I will create another ticket for this.
